examples/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,124 @@
+  else
+  C.degree=total_degree(C.eq)
+  println(C.degree)
+  end
+end
+
+######################################################
+#Compute the Jacobian Ideal of C
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_Jacobian(C::AffinePlaneCurve)
+   > Given an affine plane curve C, returns the Jacobian ideal of the defining polynomial.
+"""
+function AffinePlaneCurve_Jacobian(C::AffinePlaneCurve)
+ return jacobi_ideal(C.eq)
+end
+
+######################################################
+#To check if a point is smooth: return true if the point is a smooth point on the curve C, and false otherwise.
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_Jacobian(C::AffinePlaneCurve, P::Array{S,1}) where S<: FieldElem
+   > Given an affine plane curve C and a point P, returns false if the point is not in the zero locus of the defining equation or is a singular point of C, and true if it is a smooth point of the curve.
+"""
+function AffinePlaneCurve_issmoothpoint(C::AffinePlaneCurve{S}, P::Array{S,1}) where S<: FieldElem
+  if evaluate(C.eq, P)!=0
+     return false                         #return false if the point is not on the curve
+  else
+     J=AffinePlaneCurve_Jacobian(C)
+     L=gens(J)
+     a=evaluate(L[1],P)
+     b=evaluate(L[2],P)
+     if a==0 && b==0 
+        return false                      #return false if the point is not a smooth point of the curve
+     else
+        return true
+     end
+  end
+end
+
+######################################################
+#Compute the tangent of C at the point (x0,y0), if this point is a smooth point of C
+#For a point over QQ, it has to be written as P = [QQ(1), QQ(-1//2)]
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_tangent(C::AffinePlaneCurve, P::Array{S,1}) where S<: FieldElem
+   > Given an affine plane curve C and a point P, if P is a smooth point of C, returns the affine plane curve defined as the tangent of C at P.
+"""
+function AffinePlaneCurve_tangent(C::AffinePlaneCurve{S}, P::Array{S, 1}) where S <: FieldElem
+  if evaluate(C.eq, P)!=0
+     println("The point is not on the curve")
+  else
+     J=AffinePlaneCurve_Jacobian(C)
+     L=gens(J)
+     a=evaluate(L[1],P)
+     b=evaluate(L[2],P)
+     if a==0 && b==0
+        println("This is not a smooth point of the curve")
+     else
+        V=gens(parent(C.eq))
+        T=a*(V[1]-P[1])+b*(V[2]-P[2])
+        return AffinePlaneCurve(T)
+     end
+  end
+end
+
+####################################################
+#Union of two affine plane curves (with multiplicities)
+
+@doc Markdown.doc"""
+   AffinePlaneCurve_union(C::AffinePlaneCurve, D::AffinePlaneCurve{S}) where S<: FieldElem
+   > Given two affine plane curves C and D, returns the union of the two curves (with multiplicity).
+"""
+function AffinePlaneCurve_union(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+  F=C.eq*D.eq
+  return AffinePlaneCurve(F)
+end
+
+####################################################
+end
+using .AffinePlaneCurveModule
```

Comment:
Why does the file containing the module also use the module?

Reply:
because it is in examples and it makes development easier

Reply:
recall: this is a first attempt to geometry, in examples to make sure we can all play with it - without it begin part of "Oscar" automatically...

Reply:
That's ok. I saw it and thought it looked very interesting, hence the reason I decided to see if there was anything I could suggest. I'm really glad this is being done!